test/data/manifests/fedora-coreos-container.mpp.yaml

```diff
@@ -0,0 +1,263 @@
+version: '2'
+mpp-define-image:
+  id: image
+  #10G
+  size: '10737418240'
+  table:
+    uuid: 00000000-0000-4000-a000-000000000001
+    label: gpt
+    partitions:
+      - id: BIOS-BOOT
+        size: 2048
+        type: 21686148-6449-6E6F-744E-656564454649
+        bootable: true
+        uuid: FAC7F1FB-3E8D-4137-A512-961DE09A5549
+      - id: EFI-SYSTEM
+        size: 260096
+        type: C12A7328-F81F-11D2-BA4B-00A0C93EC93B
+        uuid: 68B2905B-DF3E-4FB3-80FA-49D1E773AA33
+      - id: boot
+        size: 786432
+        type: 0FC63DAF-8483-4772-8E79-3D69D8477DE4
+        uuid: 61B2905B-DF3E-4FB3-80FA-49D1E773AA32
+      - id: root
+        # XXX: Dynamically set this size in the future
+        size: 8388608
+        type: 0FC63DAF-8483-4772-8E79-3D69D8477DE4
+        uuid: CA7D7CCB-63ED-4C53-861C-1742536059CC
+# sources:
+#   org.osbuild.ostree:
+#     items:
+#       2c39e9b6a6b26226b626f4daf815d5307a96b2cd57463e94f7c618e312a61980:
+#         remote:
+#           url: https://kojipkgs.fedoraproject.org/ostree/repo/
+pipelines:
+  - mpp-import-pipelines:
+      path: fedora-vars.ipp.yaml
+  - mpp-import-pipeline:
+      path: fedora-build-v2.ipp.yaml
+      id: build
+    runner:
+      mpp-format-string: org.osbuild.fedora{release}
+  - name: image-tree
+    # build: name:build
+    source-epoch: 1659397331
+    stages:
+      - type: org.osbuild.ostree.init-fs
+      - type: org.osbuild.ostree.os-init
+        options:
+          osname: fedora-coreos
+      - type: org.osbuild.ostree.config
+        options:
+          repo: /ostree/repo
+          config:
+            sysroot:
+              readonly: false
+              bootloader: none
+      - type: org.osbuild.mkdir
+        options:
+          paths:
+            - path: /boot/efi
+              mode: 448
+      - type: org.osbuild.ignition
+      - type: org.osbuild.ostree.deploy
+        options:
+          osname: fedora-coreos
+          ref: localhost/fcos
```

Comment:
Might be redundant considering how the container image input defined below also specifies the ref. I'm not sure if we should make this optional though since we still have the case with no explicit inputs to the deploy stage (ie. the old example using the `org.osbuild.ostree.pull` stage). However, I think it just makes it slightly more confusing when you need to specify the same ref in two places.

Reply:
I agree. I think we can make this optional in the schema, but add a check for it in the code. i.e. validation won't catch it but at least the code would.

Reply:
Should we make `ref` optional for the ostree commit input case too? I think we should be able to get the ref from the inputs object:
```
print(inputs)
{'commits': {'path': '/run/osbuild/inputs/commits/repo', 'data': {'refs': {'2c39e9b6a6b26226b626f4daf815d5307a96b2cd57463e94f7c618e312a61980': {'ref': 'fedora/x86_64/coreos/stable'}}}}}
```

Reply:
After discussion, decided to make ref a requirement for ostree commit inputs still.